// yoastseo/build/languageProcessing/languages/pt/helpers/getClauses.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = getPortugueseClauses;
var _yoastseo = require("yoastseo");
var _Clause = _interopRequireDefault(require("../values/Clause"));
var _passiveVoiceAuxiliaries = _interopRequireDefault(require("../config/internal/passiveVoiceAuxiliaries.js"));
var _stopWords = _interopRequireDefault(require("../config/stopWords.js"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const {
  createRegexFromArray,
  getClauses
} = _yoastseo.languageProcessing;
const followingAuxiliaryExceptionWords = ["o", "a", "os", "as", "um", "ums", "uma", "umas"];
const options = {
  Clause: _Clause.default,
  stopwords: _stopWords.default,
  auxiliaries: _passiveVoiceAuxiliaries.default,
  regexes: {
    auxiliaryRegex: createRegexFromArray(_passiveVoiceAuxiliaries.default),
    stopCharacterRegex: /([:,])(?=[ \n\r\t'"+\-»«‹›<>])/ig,
    followingAuxiliaryExceptionRegex: createRegexFromArray(followingAuxiliaryExceptionWords)
  }
};

/**
 * Gets the clauses from a sentence by determining sentence breakers.
 *
 * @param {string} sentence The sentence to split up into clauses.
 *
 * @returns {Array} The array with all clauses that have an auxiliary.
 */
function getPortugueseClauses(sentence) {
  return getClauses(sentence, options);
}
