// yoastseo/build/languageProcessing/languages/hu/Researcher.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _yoastseo = require("yoastseo");
var _functionWords = _interopRequireDefault(require("./config/functionWords"));
var _transitionWords = _interopRequireDefault(require("./config/transitionWords"));
var _twoPartTransitionWords = _interopRequireDefault(require("./config/twoPartTransitionWords"));
var _firstWordExceptions = _interopRequireDefault(require("./config/firstWordExceptions"));
var _stopWords = _interopRequireDefault(require("./config/stopWords"));
var _getStemmer = _interopRequireDefault(require("./helpers/getStemmer"));
var _getClauses = _interopRequireDefault(require("./helpers/getClauses"));
var _isPassiveSentence = _interopRequireDefault(require("./helpers/isPassiveSentence"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const {
  AbstractResearcher
} = _yoastseo.languageProcessing;

// All config

// All helpers

/**
 * The researcher contains all the researches, helpers, data, and config.
 */
class Researcher extends AbstractResearcher {
  /**
   * Constructor
   * @param {Paper} paper The Paper object that is needed within the researches.
   * @constructor
   */
  constructor(paper) {
    super(paper);

    // Deletes researches that are currently not available in Hungarian.
    // When the research is available, this line should be removed.
    delete this.defaultResearches.getFleschReadingScore;
    Object.assign(this.config, {
      language: "hu",
      passiveConstructionType: "morphologicalAndPeriphrastic",
      functionWords: _functionWords.default,
      transitionWords: _transitionWords.default,
      twoPartTransitionWords: _twoPartTransitionWords.default,
      firstWordExceptions: _firstWordExceptions.default,
      stopWords: _stopWords.default
    });
    Object.assign(this.helpers, {
      getStemmer: _getStemmer.default,
      getClauses: _getClauses.default,
      isPassiveSentence: _isPassiveSentence.default
    });
  }
}
exports.default = Researcher;
