// yoastseo/build/languageProcessing/languages/cs/Researcher.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _yoastseo = require("yoastseo");
var _firstWordExceptions = _interopRequireDefault(require("./config/firstWordExceptions"));
var _stopWords = _interopRequireDefault(require("./config/stopWords"));
var _functionWords = require("./config/functionWords");
var _transitionWords = _interopRequireDefault(require("./config/transitionWords"));
var _twoPartTransitionWords = _interopRequireDefault(require("./config/twoPartTransitionWords"));
var _getClauses = _interopRequireDefault(require("./helpers/getClauses"));
var _getStemmer = _interopRequireDefault(require("./helpers/getStemmer"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const {
  AbstractResearcher
} = _yoastseo.languageProcessing;

// All config

// All helpers

/**
 * The researcher contains all the researches, helpers, data, and config.
 */
class Researcher extends AbstractResearcher {
  /**
   * Constructor
   * @param {Paper} paper The Paper object that is needed within the researches.
   * @constructor
   */
  constructor(paper) {
    super(paper);
    delete this.defaultResearches.getFleschReadingScore;
    Object.assign(this.config, {
      language: "cs",
      passiveConstructionType: "periphrastic",
      firstWordExceptions: _firstWordExceptions.default,
      stopWords: _stopWords.default,
      functionWords: _functionWords.all,
      transitionWords: _transitionWords.default,
      twoPartTransitionWords: _twoPartTransitionWords.default
    });
    Object.assign(this.helpers, {
      getClauses: _getClauses.default,
      getStemmer: _getStemmer.default
    });
  }
}
exports.default = Researcher;
