// yoastseo/build/languageProcessing/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AbstractResearcher", {
  enumerable: true,
  get: function () {
    return _AbstractResearcher.default;
  }
});
Object.defineProperty(exports, "areWordsInSentence", {
  enumerable: true,
  get: function () {
    return _areWordsInSentence.default;
  }
});
Object.defineProperty(exports, "baseStemmer", {
  enumerable: true,
  get: function () {
    return _baseStemmer.default;
  }
});
Object.defineProperty(exports, "buildFormRule", {
  enumerable: true,
  get: function () {
    return _buildFormRule.default;
  }
});
Object.defineProperty(exports, "collectMarkingsInSentence", {
  enumerable: true,
  get: function () {
    return _markWordsInSentences.collectMarkingsInSentence;
  }
});
Object.defineProperty(exports, "countMetaDescriptionLength", {
  enumerable: true,
  get: function () {
    return _countMetaDescriptionLength.default;
  }
});
Object.defineProperty(exports, "createRegexFromArray", {
  enumerable: true,
  get: function () {
    return _createRegexFromArray.default;
  }
});
Object.defineProperty(exports, "createRulesFromArrays", {
  enumerable: true,
  get: function () {
    return _createRulesFromArrays.default;
  }
});
Object.defineProperty(exports, "createSingleRuleFromArray", {
  enumerable: true,
  get: function () {
    return _createRulesFromArrays.createSingleRuleFromArray;
  }
});
Object.defineProperty(exports, "directPrecedenceException", {
  enumerable: true,
  get: function () {
    return _directPrecedenceException.default;
  }
});
exports.exceptionListHelpers = void 0;
Object.defineProperty(exports, "findMatchingEndingInArray", {
  enumerable: true,
  get: function () {
    return _findMatchingEndingInArray.default;
  }
});
Object.defineProperty(exports, "findWordFormsInString", {
  enumerable: true,
  get: function () {
    return _findKeywordFormsInString.findWordFormsInString;
  }
});
Object.defineProperty(exports, "flattenSortLength", {
  enumerable: true,
  get: function () {
    return _flattenSortLength.default;
  }
});
Object.defineProperty(exports, "getClauses", {
  enumerable: true,
  get: function () {
    return _getClauses.default;
  }
});
Object.defineProperty(exports, "getClausesSplitOnStopWords", {
  enumerable: true,
  get: function () {
    return _getClausesSplitOnStopWords.default;
  }
});
Object.defineProperty(exports, "getFieldsToMark", {
  enumerable: true,
  get: function () {
    return _getFieldsToMark.getFieldsToMark;
  }
});
Object.defineProperty(exports, "getLanguage", {
  enumerable: true,
  get: function () {
    return _getLanguage.default;
  }
});
Object.defineProperty(exports, "getSentences", {
  enumerable: true,
  get: function () {
    return _getSentences.default;
  }
});
Object.defineProperty(exports, "getWords", {
  enumerable: true,
  get: function () {
    return _getWords.default;
  }
});
exports.helpers = void 0;
Object.defineProperty(exports, "imageInText", {
  enumerable: true,
  get: function () {
    return _imageInText.default;
  }
});
Object.defineProperty(exports, "indices", {
  enumerable: true,
  get: function () {
    return _indices.default;
  }
});
exports.languageHelpers = exports.languageConfigs = void 0;
Object.defineProperty(exports, "markWordsInSentences", {
  enumerable: true,
  get: function () {
    return _markWordsInSentences.markWordsInSentences;
  }
});
Object.defineProperty(exports, "matchRegularParticiples", {
  enumerable: true,
  get: function () {
    return _matchRegularParticiples.default;
  }
});
Object.defineProperty(exports, "mergeListItems", {
  enumerable: true,
  get: function () {
    return _mergeListItems.mergeListItems;
  }
});
Object.defineProperty(exports, "nonDirectPrecedenceException", {
  enumerable: true,
  get: function () {
    return _nonDirectParticiplePrecedenceException.default;
  }
});
Object.defineProperty(exports, "normalizeHTML", {
  enumerable: true,
  get: function () {
    return _normalizeHTML.default;
  }
});
Object.defineProperty(exports, "normalizeSingle", {
  enumerable: true,
  get: function () {
    return _quotes.normalizeSingle;
  }
});
Object.defineProperty(exports, "parseSynonyms", {
  enumerable: true,
  get: function () {
    return _parseSynonyms.default;
  }
});
Object.defineProperty(exports, "precedenceException", {
  enumerable: true,
  get: function () {
    return _precedenceException.default;
  }
});
exports.regexHelpers = void 0;
Object.defineProperty(exports, "removePunctuation", {
  enumerable: true,
  get: function () {
    return _removePunctuation.default;
  }
});
Object.defineProperty(exports, "replaceDiacritics", {
  enumerable: true,
  get: function () {
    return _replaceDiacritics.default;
  }
});
exports.researches = void 0;
Object.defineProperty(exports, "sanitizeString", {
  enumerable: true,
  get: function () {
    return _sanitizeString.default;
  }
});
exports.stemHelpers = void 0;
Object.defineProperty(exports, "stripBlockTagsAtStartEnd", {
  enumerable: true,
  get: function () {
    return _stripHTMLTags.stripBlockTagsAtStartEnd;
  }
});
Object.defineProperty(exports, "stripHTMLTags", {
  enumerable: true,
  get: function () {
    return _stripHTMLTags.stripFullTags;
  }
});
Object.defineProperty(exports, "stripSpaces", {
  enumerable: true,
  get: function () {
    return _stripSpaces.default;
  }
});
Object.defineProperty(exports, "transliterate", {
  enumerable: true,
  get: function () {
    return _transliterate.default;
  }
});
Object.defineProperty(exports, "unifyAllSpaces", {
  enumerable: true,
  get: function () {
    return _unifyWhitespace.unifyAllSpaces;
  }
});
exports.values = void 0;
var _replaceDiacritics = _interopRequireDefault(require("./helpers/transliterate/replaceDiacritics"));
var _transliterate = _interopRequireDefault(require("./helpers/transliterate/transliterate"));
var _createRegexFromArray = _interopRequireDefault(require("./helpers/regex/createRegexFromArray"));
var _imageInText = _interopRequireDefault(require("./helpers/image/imageInText"));
var _stripSpaces = _interopRequireDefault(require("./helpers/sanitize/stripSpaces"));
var _baseStemmer = _interopRequireDefault(require("./helpers/morphology/baseStemmer"));
var _getWords = _interopRequireDefault(require("./helpers/word/getWords"));
var _AbstractResearcher = _interopRequireDefault(require("./AbstractResearcher"));
var _flattenSortLength = _interopRequireDefault(require("./helpers/morphology/flattenSortLength"));
var _indices = _interopRequireDefault(require("./helpers/word/indices"));
var _buildFormRule = _interopRequireDefault(require("./helpers/morphology/buildFormRule"));
var _createRulesFromArrays = _interopRequireWildcard(require("./helpers/morphology/createRulesFromArrays"));
var _matchRegularParticiples = _interopRequireDefault(require("./helpers/passiveVoice/periphrastic/matchRegularParticiples"));
var _directPrecedenceException = _interopRequireDefault(require("./helpers/passiveVoice/periphrastic/directPrecedenceException"));
var _precedenceException = _interopRequireDefault(require("./helpers/passiveVoice/periphrastic/precedenceException"));
var _nonDirectParticiplePrecedenceException = _interopRequireDefault(require("./helpers/passiveVoice/periphrastic/freeAuxiliaryParticipleOrder/nonDirectParticiplePrecedenceException"));
var _findMatchingEndingInArray = _interopRequireDefault(require("./helpers/morphology/findMatchingEndingInArray"));
var regexHelpers = _interopRequireWildcard(require("./helpers/morphology/regexHelpers"));
exports.regexHelpers = regexHelpers;
var exceptionListHelpers = _interopRequireWildcard(require("./helpers/morphology/exceptionListHelpers"));
exports.exceptionListHelpers = exceptionListHelpers;
var stemHelpers = _interopRequireWildcard(require("./helpers/morphology/stemHelpers"));
exports.stemHelpers = stemHelpers;
var values = _interopRequireWildcard(require("./values"));
exports.values = values;
var _areWordsInSentence = _interopRequireDefault(require("./helpers/word/areWordsInSentence"));
var _getClauses = _interopRequireDefault(require("./helpers/passiveVoice/periphrastic/getClauses"));
var _getClausesSplitOnStopWords = _interopRequireDefault(require("./helpers/passiveVoice/periphrastic/freeAuxiliaryParticipleOrder/getClausesSplitOnStopWords"));
var _stripHTMLTags = require("./helpers/sanitize/stripHTMLTags");
var _sanitizeString = _interopRequireDefault(require("./helpers/sanitize/sanitizeString"));
var _unifyWhitespace = require("./helpers/sanitize/unifyWhitespace");
var _removePunctuation = _interopRequireDefault(require("./helpers/sanitize/removePunctuation"));
var _normalizeHTML = _interopRequireDefault(require("./helpers/html/normalizeHTML"));
var _countMetaDescriptionLength = _interopRequireDefault(require("./helpers/word/countMetaDescriptionLength"));
var _getLanguage = _interopRequireDefault(require("./helpers/language/getLanguage"));
var _getSentences = _interopRequireDefault(require("./helpers/sentence/getSentences"));
var _getFieldsToMark = require("./helpers/html/getFieldsToMark");
var _quotes = require("./helpers/sanitize/quotes");
var _parseSynonyms = _interopRequireDefault(require("./helpers/sanitize/parseSynonyms"));
var _mergeListItems = require("./helpers/sanitize/mergeListItems");
var _findKeywordFormsInString = require("./helpers/match/findKeywordFormsInString");
var _markWordsInSentences = require("./helpers/word/markWordsInSentences");
var helpers = _interopRequireWildcard(require("./helpers"));
exports.helpers = helpers;
var researches = _interopRequireWildcard(require("./researches"));
exports.researches = researches;
var _checkIfWordIsComplex = _interopRequireDefault(require("./languages/en/helpers/checkIfWordIsComplex"));
var _checkIfWordIsComplex2 = _interopRequireDefault(require("./languages/de/helpers/checkIfWordIsComplex"));
var _checkIfWordIsComplex3 = _interopRequireDefault(require("./languages/es/helpers/checkIfWordIsComplex"));
var _checkIfWordIsComplex4 = _interopRequireDefault(require("./languages/fr/helpers/checkIfWordIsComplex"));
var _wordComplexity = _interopRequireDefault(require("./languages/en/config/wordComplexity"));
var _wordComplexity2 = _interopRequireDefault(require("./languages/de/config/wordComplexity"));
var _wordComplexity3 = _interopRequireDefault(require("./languages/es/config/wordComplexity"));
var _wordComplexity4 = _interopRequireDefault(require("./languages/fr/config/wordComplexity"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const languageHelpers = exports.languageHelpers = {
  wordComplexityHelperEnglish: _checkIfWordIsComplex.default,
  wordComplexityHelperGerman: _checkIfWordIsComplex2.default,
  wordComplexityHelperSpanish: _checkIfWordIsComplex3.default,
  wordComplexityHelperFrench: _checkIfWordIsComplex4.default
};
const languageConfigs = exports.languageConfigs = {
  wordComplexityConfigEnglish: _wordComplexity.default,
  wordComplexityConfigGerman: _wordComplexity2.default,
  wordComplexityConfigSpanish: _wordComplexity3.default,
  wordComplexityConfigFrench: _wordComplexity4.default
};
