// yoastseo/build/languageProcessing/languages/sk/helpers/getClauses.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = getSlovakClauses;
var _yoastseo = require("yoastseo");
var _Clause = _interopRequireDefault(require("../values/Clause"));
var _passiveVoiceAuxiliaries = _interopRequireDefault(require("../config/internal/passiveVoiceAuxiliaries.js"));
var _stopWords = _interopRequireDefault(require("../config/stopWords.js"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const {
  getClausesSplitOnStopWords,
  createRegexFromArray
} = _yoastseo.languageProcessing;
const options = {
  Clause: _Clause.default,
  regexes: {
    auxiliaryRegex: createRegexFromArray(_passiveVoiceAuxiliaries.default),
    stopCharacterRegex: /([:,])(?=[ \n\r\t'"+\-»«‹›<>])/ig,
    stopwordRegex: createRegexFromArray(_stopWords.default)
  }
};

/**
 * Gets the clauses from a sentence by determining sentence breakers.
 *
 * @param {string} sentence The sentence to split up into clauses.
 *
 * @returns {Array} The array with all clauses that have an auxiliary.
 */
function getSlovakClauses(sentence) {
  return getClausesSplitOnStopWords(sentence, options);
}
