// yoastseo/build/config/presenter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = _default;
var _i18n = require("@wordpress/i18n");
/**
 * Returns the configuration used for score ratings and the AssessorPresenter.
 * @returns {Object} The config object.
 */
function _default() {
  const contentOptimizationLabel = (0, _i18n.__)("Content optimization:", "wordpress-seo");
  return {
    feedback: {
      className: "na",
      screenReaderText: (0, _i18n.__)("Feedback", "wordpress-seo"),
      fullText: `${contentOptimizationLabel} ${(0, _i18n.__)("Has feedback", "wordpress-seo")}`,
      screenReaderReadabilityText: ""
    },
    bad: {
      className: "bad",
      screenReaderText: (0, _i18n.__)("Needs improvement", "wordpress-seo"),
      fullText: `${contentOptimizationLabel} ${(0, _i18n.__)("Needs improvement", "wordpress-seo")}`,
      screenReaderReadabilityText: (0, _i18n.__)("Needs improvement", "wordpress-seo")
    },
    ok: {
      className: "ok",
      screenReaderText: (0, _i18n.__)("OK SEO score", "wordpress-seo"),
      fullText: `${contentOptimizationLabel} ${(0, _i18n.__)("OK SEO score", "wordpress-seo")}`,
      screenReaderReadabilityText: (0, _i18n.__)("OK", "wordpress-seo")
    },
    good: {
      className: "good",
      screenReaderText: (0, _i18n.__)("Good SEO score", "wordpress-seo"),
      fullText: `${contentOptimizationLabel} ${(0, _i18n.__)("Good SEO score", "wordpress-seo")}`,
      screenReaderReadabilityText: (0, _i18n.__)("Good", "wordpress-seo")
    }
  };
}
