// yoastseo/build/scoring/productPages/contentAssessor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _util = require("util");
var _yoastseo = require("yoastseo");
const {
  createAnchorOpeningTag
} = _yoastseo.helpers;
const {
  ParagraphTooLongAssessment,
  SentenceLengthInTextAssessment,
  SubheadingDistributionTooLongAssessment,
  TransitionWordsAssessment,
  PassiveVoiceAssessment,
  TextPresenceAssessment,
  ListAssessment
} = _yoastseo.assessments.readability;

/**
 * Creates the Assessor
 *
 * @param {object} researcher   The researcher to use for the analysis.
 * @param {Object} options      The options for this assessor.
 *
 * @constructor
 */
const ProductContentAssessor = function (researcher, options) {
  _yoastseo.Assessor.call(this, researcher, options);
  this.type = "productContentAssessor";
  this._assessments = [new SubheadingDistributionTooLongAssessment({
    shouldNotAppearInShortText: true,
    urlTitle: createAnchorOpeningTag(options.subheadingUrlTitle),
    urlCallToAction: createAnchorOpeningTag(options.subheadingCTAUrl)
  }), new ParagraphTooLongAssessment({
    parameters: {
      recommendedLength: 70,
      maximumRecommendedLength: 100
    },
    urlTitle: createAnchorOpeningTag(options.paragraphUrlTitle),
    urlCallToAction: createAnchorOpeningTag(options.paragraphCTAUrl)
  }, true), new SentenceLengthInTextAssessment({
    slightlyTooMany: 20,
    farTooMany: 25,
    urlTitle: createAnchorOpeningTag(options.sentenceLengthUrlTitle),
    urlCallToAction: createAnchorOpeningTag(options.sentenceLengthCTAUrl)
  }, false, true), new TransitionWordsAssessment({
    urlTitle: createAnchorOpeningTag(options.transitionWordsUrlTitle),
    urlCallToAction: createAnchorOpeningTag(options.transitionWordsCTAUrl)
  }), new PassiveVoiceAssessment({
    urlTitle: createAnchorOpeningTag(options.passiveVoiceUrlTitle),
    urlCallToAction: createAnchorOpeningTag(options.passiveVoiceCTAUrl)
  }), new TextPresenceAssessment({
    urlTitle: createAnchorOpeningTag(options.textPresenceUrlTitle),
    urlCallToAction: createAnchorOpeningTag(options.textPresenceCTAUrl)
  }), new ListAssessment({
    urlTitle: createAnchorOpeningTag(options.listsUrlTitle),
    urlCallToAction: createAnchorOpeningTag(options.listsCTAUrl)
  })];
};
(0, _util.inherits)(ProductContentAssessor, _yoastseo.ContentAssessor);
var _default = exports.default = ProductContentAssessor;
