// yoastseo/build/languageProcessing/languages/ca/Researcher.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _yoastseo = require("yoastseo");
var _transitionWords = _interopRequireDefault(require("./config/transitionWords"));
var _twoPartTransitionWords = _interopRequireDefault(require("./config/twoPartTransitionWords"));
var _sentenceLength = _interopRequireDefault(require("./config/sentenceLength"));
var _getStemmer = _interopRequireDefault(require("./helpers/getStemmer"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const {
  AbstractResearcher
} = _yoastseo.languageProcessing;

// All config

// All helpers

/**
 * The researcher contains all the researches, helpers, data, and config.
 */
class Researcher extends AbstractResearcher {
  /**
   * Constructor
   * @param {Paper} paper The Paper object that is needed within the researches.
   * @constructor
   */
  constructor(paper) {
    super(paper);

    // Deletes researches that are currently not available in Catalan.
    // When the research is available, this line should be removed.
    delete this.defaultResearches.getFleschReadingScore;
    delete this.defaultResearches.getPassiveVoiceResult;
    delete this.defaultResearches.getSentenceBeginnings;
    delete this.defaultResearches.functionWordsInKeyphrase;
    Object.assign(this.config, {
      language: "ca",
      functionWords: [],
      transitionWords: _transitionWords.default,
      twoPartTransitionWords: _twoPartTransitionWords.default,
      sentenceLength: _sentenceLength.default
    });
    Object.assign(this.helpers, {
      getStemmer: _getStemmer.default
    });
  }
}
exports.default = Researcher;
