// yoastseo/build/languageProcessing/languages/ru/Researcher.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _yoastseo = require("yoastseo");
var _firstWordExceptions = _interopRequireDefault(require("./config/firstWordExceptions"));
var _functionWords = require("./config/functionWords");
var _transitionWords = _interopRequireDefault(require("./config/transitionWords"));
var _twoPartTransitionWords = _interopRequireDefault(require("./config/twoPartTransitionWords"));
var _syllables = _interopRequireDefault(require("./config/syllables.json"));
var _fleschReadingEaseScores = _interopRequireDefault(require("./config/fleschReadingEaseScores"));
var _sentenceLength = _interopRequireDefault(require("./config/sentenceLength"));
var _getStemmer = _interopRequireDefault(require("./helpers/getStemmer"));
var _isPassiveSentence = _interopRequireDefault(require("./helpers/isPassiveSentence"));
var _calculateFleschReadingScore = _interopRequireDefault(require("./helpers/calculateFleschReadingScore"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const {
  AbstractResearcher
} = _yoastseo.languageProcessing;

// All config

// All helpers

/**
 * The researcher contains all the researches, helpers, data, and config.
 */
class Researcher extends AbstractResearcher {
  /**
   * Constructor
   * @param {Paper} paper The Paper object that is needed within the researches.
   * @constructor
   */
  constructor(paper) {
    super(paper);
    Object.assign(this.config, {
      language: "ru",
      passiveConstructionType: "morphological",
      firstWordExceptions: _firstWordExceptions.default,
      functionWords: _functionWords.all,
      transitionWords: _transitionWords.default,
      twoPartTransitionWords: _twoPartTransitionWords.default,
      syllables: _syllables.default,
      fleschReadingEaseScores: _fleschReadingEaseScores.default,
      sentenceLength: _sentenceLength.default
    });
    Object.assign(this.helpers, {
      getStemmer: _getStemmer.default,
      isPassiveSentence: _isPassiveSentence.default,
      fleschReadingScore: _calculateFleschReadingScore.default
    });
  }
}
exports.default = Researcher;
