// yoastseo/build/scoring/cornerstone/contentAssessor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _assessor = _interopRequireDefault(require("../assessor.js"));
var _contentAssessor = _interopRequireDefault(require("../contentAssessor"));
var _ParagraphTooLongAssessment = _interopRequireDefault(require("../assessments/readability/ParagraphTooLongAssessment.js"));
var _SentenceLengthInTextAssessment = _interopRequireDefault(require("../assessments/readability/SentenceLengthInTextAssessment.js"));
var _SubheadingDistributionTooLongAssessment = _interopRequireDefault(require("../assessments/readability/SubheadingDistributionTooLongAssessment.js"));
var _TransitionWordsAssessment = _interopRequireDefault(require("../assessments/readability/TransitionWordsAssessment.js"));
var _PassiveVoiceAssessment = _interopRequireDefault(require("../assessments/readability/PassiveVoiceAssessment.js"));
var _SentenceBeginningsAssessment = _interopRequireDefault(require("../assessments/readability/SentenceBeginningsAssessment.js"));
var _TextPresenceAssessment = _interopRequireDefault(require("../assessments/readability/TextPresenceAssessment.js"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/*
 Temporarily disabled:

 var sentenceLengthInDescription = require( "./assessments/readability/sentenceLengthInDescriptionAssessment.js" );
 */

/**
 * Creates the Assessor
 *
 * @param {object} researcher       The researcher used for the analysis.
 * @param {Object} options          The options for this assessor.
 * @param {Object} options.marker   The marker to pass the list of marks to.
 *
 * @constructor
 */
const CornerStoneContentAssessor = function (researcher, options = {}) {
  _assessor.default.call(this, researcher, options);
  this.type = "cornerstoneContentAssessor";
  this._assessments = [new _SubheadingDistributionTooLongAssessment.default({
    parameters: {
      slightlyTooMany: 250,
      farTooMany: 300,
      recommendedMaximumLength: 250
    },
    applicableIfTextLongerThan: 250,
    cornerstoneContent: true
  }), new _ParagraphTooLongAssessment.default(), new _SentenceLengthInTextAssessment.default({
    slightlyTooMany: 20,
    farTooMany: 25
  }, true), new _TransitionWordsAssessment.default(), new _PassiveVoiceAssessment.default(), new _TextPresenceAssessment.default(), new _SentenceBeginningsAssessment.default()
  // Temporarily disabled: wordComplexity,
  ];
};
require("util").inherits(CornerStoneContentAssessor, _contentAssessor.default);
var _default = exports.default = CornerStoneContentAssessor;
