// yoastseo/build/scoring/cornerstone/relatedKeywordAssessor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _util = require("util");
var _assessor = _interopRequireDefault(require("../assessor.js"));
var _IntroductionKeywordAssessment = _interopRequireDefault(require("../assessments/seo/IntroductionKeywordAssessment.js"));
var _KeyphraseLengthAssessment = _interopRequireDefault(require("../assessments/seo/KeyphraseLengthAssessment.js"));
var _KeywordDensityAssessment = _interopRequireDefault(require("../assessments/seo/KeywordDensityAssessment.js"));
var _MetaDescriptionKeywordAssessment = _interopRequireDefault(require("../assessments/seo/MetaDescriptionKeywordAssessment.js"));
var _TextCompetingLinksAssessment = _interopRequireDefault(require("../assessments/seo/TextCompetingLinksAssessment.js"));
var _FunctionWordsInKeyphraseAssessment = _interopRequireDefault(require("../assessments/seo/FunctionWordsInKeyphraseAssessment"));
var _KeyphraseInImageTextAssessment = _interopRequireDefault(require("../assessments/seo/KeyphraseInImageTextAssessment"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/**
 * Creates the Assessor
 *
 * @param {Researcher} researcher    The researcher used for the analysis.
 * @param {Object?} options          The options for this assessor.
 * @param {Function} options.marker  The marker to pass the list of marks to.
 *
 * @constructor
 */
const relatedKeywordAssessor = function (researcher, options) {
  _assessor.default.call(this, researcher, options);
  this.type = "cornerstoneRelatedKeywordAssessor";
  this._assessments = [new _IntroductionKeywordAssessment.default(), new _KeyphraseLengthAssessment.default({
    isRelatedKeyphrase: true
  }), new _KeywordDensityAssessment.default(), new _MetaDescriptionKeywordAssessment.default(), new _TextCompetingLinksAssessment.default(), new _FunctionWordsInKeyphraseAssessment.default(), new _KeyphraseInImageTextAssessment.default({
    scores: {
      withAltNonKeyword: 3,
      withAlt: 3,
      noAlt: 3
    }
  })];
};
(0, _util.inherits)(relatedKeywordAssessor, _assessor.default);
var _default = exports.default = relatedKeywordAssessor;
