// yoastseo/build/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AnalysisWebWorker", {
  enumerable: true,
  get: function () {
    return _worker.AnalysisWebWorker;
  }
});
Object.defineProperty(exports, "AnalysisWorkerWrapper", {
  enumerable: true,
  get: function () {
    return _worker.AnalysisWorkerWrapper;
  }
});
Object.defineProperty(exports, "App", {
  enumerable: true,
  get: function () {
    return _app.default;
  }
});
Object.defineProperty(exports, "Assessment", {
  enumerable: true,
  get: function () {
    return _assessment.default;
  }
});
Object.defineProperty(exports, "AssessmentResult", {
  enumerable: true,
  get: function () {
    return _AssessmentResult.default;
  }
});
Object.defineProperty(exports, "Assessor", {
  enumerable: true,
  get: function () {
    return _assessor.default;
  }
});
Object.defineProperty(exports, "ContentAssessor", {
  enumerable: true,
  get: function () {
    return _contentAssessor.default;
  }
});
Object.defineProperty(exports, "DIFFICULTY", {
  enumerable: true,
  get: function () {
    return _getFleschReadingScore.DIFFICULTY;
  }
});
Object.defineProperty(exports, "Factory", {
  enumerable: true,
  get: function () {
    return _factory.default;
  }
});
Object.defineProperty(exports, "Paper", {
  enumerable: true,
  get: function () {
    return _Paper.default;
  }
});
Object.defineProperty(exports, "Pluggable", {
  enumerable: true,
  get: function () {
    return _pluggable.default;
  }
});
Object.defineProperty(exports, "SeoAssessor", {
  enumerable: true,
  get: function () {
    return _seoAssessor.default;
  }
});
Object.defineProperty(exports, "TaxonomyAssessor", {
  enumerable: true,
  get: function () {
    return _taxonomyAssessor.default;
  }
});
exports.config = exports.bundledPlugins = exports.assessments = void 0;
Object.defineProperty(exports, "createWorker", {
  enumerable: true,
  get: function () {
    return _worker.createWorker;
  }
});
exports.values = exports.markers = exports.languageProcessing = exports.interpreters = exports.helpers = exports.default = void 0;
var _worker = require("./worker");
var assessments = _interopRequireWildcard(require("./scoring/assessments"));
exports.assessments = assessments;
var bundledPlugins = _interopRequireWildcard(require("./bundledPlugins"));
exports.bundledPlugins = bundledPlugins;
var helpers = _interopRequireWildcard(require("./helpers"));
exports.helpers = helpers;
var markers = _interopRequireWildcard(require("./markers"));
exports.markers = markers;
var interpreters = _interopRequireWildcard(require("./scoring/interpreters"));
exports.interpreters = interpreters;
var config = _interopRequireWildcard(require("./config"));
exports.config = config;
var languageProcessing = _interopRequireWildcard(require("./languageProcessing"));
exports.languageProcessing = languageProcessing;
var values = _interopRequireWildcard(require("./values"));
exports.values = values;
var _app = _interopRequireDefault(require("./app"));
var _assessor = _interopRequireDefault(require("./scoring/assessor"));
var _contentAssessor = _interopRequireDefault(require("./scoring/contentAssessor"));
var _seoAssessor = _interopRequireDefault(require("./scoring/seoAssessor"));
var _taxonomyAssessor = _interopRequireDefault(require("./scoring/taxonomyAssessor"));
var _pluggable = _interopRequireDefault(require("./pluggable"));
var _Paper = _interopRequireDefault(require("./values/Paper"));
var _AssessmentResult = _interopRequireDefault(require("./values/AssessmentResult"));
var _assessment = _interopRequireDefault(require("./scoring/assessments/assessment"));
var _getFleschReadingScore = require("./languageProcessing/researches/getFleschReadingScore");
var _factory = _interopRequireDefault(require("./helpers/factory"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/*
 * Everything exported here is put on the `yoast.analysis` global in the plugin.
 */
/*
 * Used for backwards compatibility reasons.
 * For new exports, please add it as a named dependency above instead.
 */
var _default = exports.default = {
  App: _app.default,
  Assessor: _assessor.default,
  ContentAssessor: _contentAssessor.default,
  TaxonomyAssessor: _taxonomyAssessor.default,
  Pluggable: _pluggable.default,
  Paper: _Paper.default,
  AssessmentResult: _AssessmentResult.default,
  AnalysisWebWorker: _worker.AnalysisWebWorker,
  AnalysisWorkerWrapper: _worker.AnalysisWorkerWrapper,
  createWorker: _worker.createWorker,
  assessments,
  bundledPlugins,
  config,
  helpers,
  markers,
  interpreters,
  languageProcessing,
  values
};
