// yoastseo/build/scoring/taxonomyAssessor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getTextLengthAssessment = exports.default = void 0;
var _util = require("util");
var _IntroductionKeywordAssessment = _interopRequireDefault(require("./assessments/seo/IntroductionKeywordAssessment"));
var _KeyphraseLengthAssessment = _interopRequireDefault(require("./assessments/seo/KeyphraseLengthAssessment"));
var _KeywordDensityAssessment = _interopRequireDefault(require("./assessments/seo/KeywordDensityAssessment"));
var _MetaDescriptionKeywordAssessment = _interopRequireDefault(require("./assessments/seo/MetaDescriptionKeywordAssessment"));
var _KeyphraseInSEOTitleAssessment = _interopRequireDefault(require("./assessments/seo/KeyphraseInSEOTitleAssessment"));
var _UrlKeywordAssessment = _interopRequireDefault(require("./assessments/seo/UrlKeywordAssessment"));
var _assessor = _interopRequireDefault(require("./assessor"));
var _MetaDescriptionLengthAssessment = _interopRequireDefault(require("./assessments/seo/MetaDescriptionLengthAssessment"));
var _TextLengthAssessment = _interopRequireDefault(require("./assessments/seo/TextLengthAssessment"));
var _PageTitleWidthAssessment = _interopRequireDefault(require("./assessments/seo/PageTitleWidthAssessment"));
var _FunctionWordsInKeyphraseAssessment = _interopRequireDefault(require("./assessments/seo/FunctionWordsInKeyphraseAssessment"));
var _SingleH1Assessment = _interopRequireDefault(require("./assessments/seo/SingleH1Assessment"));
var _helpers = require("../helpers");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/**
 * Returns the text length assessment to use.
 *
 * @returns {TextLengthAssessment} The text length assessment (with taxonomy configuration) to use.
 */
const getTextLengthAssessment = function () {
  // Export so it can be used in tests.
  return new _TextLengthAssessment.default({
    recommendedMinimum: 30,
    slightlyBelowMinimum: 10,
    veryFarBelowMinimum: 1,
    urlTitle: (0, _helpers.createAnchorOpeningTag)("https://yoa.st/34j"),
    urlCallToAction: (0, _helpers.createAnchorOpeningTag)("https://yoa.st/34k"),
    customContentType: "taxonomyAssessor"
  });
};

/**
 * Creates the Assessor used for taxonomy pages.
 *
 * @param {Researcher} researcher   The researcher used for the analysis.
 * @param {Object?} options         The options for this assessor.
 * @constructor
 */
exports.getTextLengthAssessment = getTextLengthAssessment;
const TaxonomyAssessor = function (researcher, options) {
  _assessor.default.call(this, researcher, options);
  this.type = "taxonomyAssessor";
  this._assessments = [new _IntroductionKeywordAssessment.default(), new _KeyphraseLengthAssessment.default(), new _KeywordDensityAssessment.default(), new _MetaDescriptionKeywordAssessment.default(), new _MetaDescriptionLengthAssessment.default(), getTextLengthAssessment(), new _KeyphraseInSEOTitleAssessment.default(), new _PageTitleWidthAssessment.default({
    scores: {
      widthTooShort: 9
    }
  }, true), new _UrlKeywordAssessment.default(), new _FunctionWordsInKeyphraseAssessment.default(), new _SingleH1Assessment.default()];
};
(0, _util.inherits)(TaxonomyAssessor, _assessor.default);
var _default = exports.default = TaxonomyAssessor;
