// yoastseo/build/languageProcessing/languages/ar/Researcher.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _yoastseo = require("yoastseo");
var _firstWordExceptions = _interopRequireDefault(require("./config/firstWordExceptions"));
var _functionWords = require("./config/functionWords");
var _transitionWords = _interopRequireDefault(require("./config/transitionWords"));
var _twoPartTransitionWords = _interopRequireDefault(require("./config/twoPartTransitionWords"));
var _prefixedFunctionWords = require("./config/prefixedFunctionWords");
var _createBasicWordForms = require("./helpers/createBasicWordForms");
var _getStemmer = _interopRequireDefault(require("./helpers/getStemmer"));
var _isPassiveSentence = _interopRequireDefault(require("./helpers/isPassiveSentence"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const {
  AbstractResearcher
} = _yoastseo.languageProcessing;

// All config

// All helpers

/**
 * The researcher contains all the researches, helpers, data, and config.
 */
class Researcher extends AbstractResearcher {
  /**
   * Constructor
   * @param {Paper} paper The Paper object that is needed within the researches.
   * @constructor
   */
  constructor(paper) {
    super(paper);
    delete this.defaultResearches.getFleschReadingScore;
    Object.assign(this.config, {
      language: "ar",
      passiveConstructionType: "morphological",
      firstWordExceptions: _firstWordExceptions.default,
      functionWords: _functionWords.all,
      transitionWords: _transitionWords.default,
      twoPartTransitionWords: _twoPartTransitionWords.default,
      prefixedFunctionWordsRegex: _prefixedFunctionWords.PREFIXED_FUNCTION_WORDS_REGEX
    });
    Object.assign(this.helpers, {
      createBasicWordForms: _createBasicWordForms.createBasicWordForms,
      getStemmer: _getStemmer.default,
      isPassiveSentence: _isPassiveSentence.default
    });
  }
}
exports.default = Researcher;
