// yoastseo/build/languageProcessing/languages/ja/Researcher.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _yoastseo = require("yoastseo");
var _matchTextWithWord = _interopRequireDefault(require("./helpers/matchTextWithWord"));
var _getWords = _interopRequireDefault(require("./helpers/getWords"));
var _customGetStemmer = _interopRequireDefault(require("./helpers/customGetStemmer"));
var _wordsCharacterCount = _interopRequireDefault(require("./helpers/wordsCharacterCount"));
var _countCharacters = _interopRequireDefault(require("./helpers/countCharacters"));
var _matchTransitionWords = _interopRequireDefault(require("./helpers/matchTransitionWords"));
var _getContentWords = _interopRequireDefault(require("./helpers/getContentWords"));
var _memoizedSentenceTokenizer = _interopRequireDefault(require("./helpers/memoizedSentenceTokenizer"));
var _splitIntoTokensCustom = _interopRequireDefault(require("./helpers/splitIntoTokensCustom"));
var _firstWordExceptions = _interopRequireDefault(require("./config/firstWordExceptions"));
var _functionWords = _interopRequireDefault(require("./config/functionWords"));
var _transitionWords = _interopRequireDefault(require("./config/transitionWords"));
var _topicLength = _interopRequireDefault(require("./config/topicLength"));
var _textLength = _interopRequireDefault(require("./config/textLength"));
var _paragraphLength = _interopRequireDefault(require("./config/paragraphLength"));
var _assessmentApplicabilityCharacterCount = _interopRequireDefault(require("./config/assessmentApplicabilityCharacterCount"));
var _sentenceLength = _interopRequireDefault(require("./config/sentenceLength"));
var _subheadingsTooLong = _interopRequireDefault(require("./config/subheadingsTooLong"));
var _keyphraseLength = _interopRequireDefault(require("./config/keyphraseLength"));
var _metaDescriptionLength = _interopRequireDefault(require("./config/metaDescriptionLength"));
var _getWordForms = _interopRequireDefault(require("./customResearches/getWordForms"));
var _getKeyphraseLength = _interopRequireDefault(require("./customResearches/getKeyphraseLength"));
var _textLength2 = _interopRequireDefault(require("./customResearches/textLength"));
var _findKeyphraseInSEOTitle = _interopRequireDefault(require("./customResearches/findKeyphraseInSEOTitle"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const {
  AbstractResearcher
} = _yoastseo.languageProcessing;

// All helpers

// All config

// All custom researches

/**
 * The researcher contains all the researches, helpers, data, and config.
 */
class Researcher extends AbstractResearcher {
  /**
   * Constructor
   * @param {Paper} paper The Paper object that is needed within the researches.
   * @constructor
   */
  constructor(paper) {
    super(paper);

    // Deletes researches that are not available for languages that we haven't supported yet.
    delete this.defaultResearches.getFleschReadingScore;
    delete this.defaultResearches.getPassiveVoiceResult;
    delete this.defaultResearches.keywordCountInSlug;
    Object.assign(this.config, {
      language: "ja",
      firstWordExceptions: _firstWordExceptions.default,
      functionWords: _functionWords.default,
      transitionWords: _transitionWords.default,
      topicLength: _topicLength.default,
      textLength: _textLength.default,
      paragraphLength: _paragraphLength.default,
      assessmentApplicability: _assessmentApplicabilityCharacterCount.default,
      sentenceLength: _sentenceLength.default,
      keyphraseLength: _keyphraseLength.default,
      subheadingsTooLong: _subheadingsTooLong.default,
      countCharacters: true,
      metaDescriptionLength: _metaDescriptionLength.default
    });
    Object.assign(this.helpers, {
      matchWordCustomHelper: _matchTextWithWord.default,
      getWordsCustomHelper: _getWords.default,
      getContentWords: _getContentWords.default,
      customGetStemmer: _customGetStemmer.default,
      wordsCharacterCount: _wordsCharacterCount.default,
      customCountLength: _countCharacters.default,
      matchTransitionWordsHelper: _matchTransitionWords.default,
      memoizedTokenizer: _memoizedSentenceTokenizer.default,
      splitIntoTokensCustom: _splitIntoTokensCustom.default
    });
    Object.assign(this.defaultResearches, {
      morphology: _getWordForms.default,
      keyphraseLength: _getKeyphraseLength.default,
      wordCountInText: _textLength2.default,
      findKeyphraseInSEOTitle: _findKeyphraseInSEOTitle.default
    });
  }
}
exports.default = Researcher;
